src/NmeaParser/Nmea/NmeaMessage.cs

```
@@ -304,6 +310,68 @@ internal static TimeSpan StringToTimeSpan(string value)
+
+            int fullYear = (year < 80) ? (2000 + year) : (1900 + year);
+
+            // 3. Parse time (hhmmss.ss)
+            //    - The fractional seconds might be optional or might have variable length.
+            //    - Parse hours, minutes, and then the part after the decimal.
+            //    - Time can be "hhmmss", or "hhmmss.sss", etc.
+            // Example: "102939.00" => hh=10, mm=29, ss=39, fraction=0
+            if (timeField.Length < 6) return null;
+            if (!int.TryParse(timeField.Substring(0, 2), out int hours)) return null;     // hh
+            if (!int.TryParse(timeField.Substring(2, 2), out int minutes)) return null;   // mm
+
+            // The seconds part can have a decimal fraction
+            // So we split around the decimal point:
+            double secondsDouble;
+            if (timeField.Length > 4)
+            {
+                string secString = timeField.Substring(4);  // "39.00 in example above"
+                if (!double.TryParse(secString,
+                                     System.Globalization.NumberStyles.Float,
+                                     System.Globalization.CultureInfo.InvariantCulture,
+                                     out secondsDouble))
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                secondsDouble = 0.0;
+            }
+            int seconds = (int)secondsDouble;
+            int milliseconds = (int)((secondsDouble - seconds) * 1000.0);
+
+            // 4. Construct the UTC DateTime
+            try
+            {
+                var result = new DateTime(fullYear, month, day, hours, minutes, seconds, milliseconds, DateTimeKind.Utc);
+                return result;
+            }
+            catch
+            {
+                // If the date/time is invalid (e.g. month=13, day=32), we return null or handle differently
+                return null;
```

Generally if the message is invalid, we fail the parsing, rather than ignore it.

---

Reverted.